apps/opik-backend/src/main/java/com/comet/opik/infrastructure/auth/RemoteAuthService.java

```diff
@@ -145,13 +145,12 @@ private AuthResponse verifyResponse(Response response) {
 
             return authResponse;
         } else if (response.getStatus() == Response.Status.UNAUTHORIZED.getStatusCode()) {
-            throw new ClientErrorException(NOT_ALLOWED_TO_ACCESS_WORKSPACE,
-                    Response.Status.UNAUTHORIZED);
+            throw new ClientErrorException(NOT_ALLOWED_TO_ACCESS_WORKSPACE, Response.Status.UNAUTHORIZED);
         } else if (response.getStatus() == Response.Status.FORBIDDEN.getStatusCode()) {
             throw new ClientErrorException("User has bot permission to the workspace", Response.Status.FORBIDDEN);
         } else if (response.getStatusInfo().getFamily() == Response.Status.Family.SERVER_ERROR) {
             log.error("Error while authenticating user");
-            throw new ClientErrorException(Response.Status.INTERNAL_SERVER_ERROR);
+            throw new ClientErrorException(Response.Status.BAD_REQUEST);
```

Comment:
There's another potential occurrence of this below. Please fix it as well.

Reply:
You should do the reverse fix. 

From the external user perspective, if this internal call towards EM failed with a `SERVER_ERROR` we should still return an `INTERNAL_SERVER_ERROR` to the user, not a `BAD_REQUEST`. 

But that should be the legitimate 500 code, instead of a 500 coming because instantiating the exception object here throws a `IllegalArgumentException` due to the miss-match  between the `ClientErrorException` and the former `INTERNAL_SERVER_ERROR. 

In summary, the reverse fix should be changing this to an `InternalServerErrorException`.

Finally, the fact that there are two occurrences of this problem is caused by some spaghetti code. Basically this last `else if` statement is the same as the code below outside the if-else chain

You should remove that duplication, either leave one last `else` statement or leave the code outside the if-else chain, but not both. Basically, we just need one piece of code as "catch-all" internal server exception, not two pieces doing the same.

Reply:
I was going to ask this why 400 instead of 500

Reply:
The JIRA ticket title is misleading as it says:

> When workspace is absent authentication is returning 500

That's correct in my view, but for that first we need to improve the real Authentication endpoint to return a 4XX in those cases, so this endpoint also propagates a 4XX.

Then, there's the underlaying issue on this PR to fix the bubbling up IllegalArgumentException that ends up in a 5XX, by still returning a 5XX, but coming from a InternalServerErrorException.

Reply:
Agree with all of the above. Fixed.